### PR TITLE
Stig tmp home

### PR DESCRIPTION
**Pull request by StigNorland**
added doiPrefix & doiAgencyName to CustomerDao & CustomerDto
and modified builders and test to comply

### CODE REVIEW

File: customer-commons/src/main/java/no/unit/nva/customer/model/CustomerDao.java [hunk 1]
```diff
@@ -84,7 +88,7 @@ public int hashCode() {
         return Objects.hash(getIdentifier(), getCreatedDate(), getModifiedDate(), getName(), getDisplayName(),
             getShortName(), getArchiveName(), getCname(), getInstitutionDns(),
             getFeideOrganizationDomain(), getCristinId(), getCustomerOf(), getVocabularies(),
-            getRorId(), getPublicationWorkflow());
+            getRorId(), getPublicationWorkflow(), getDoiPrefix(),getDoiAgencyName());
```
Comment (ketilaa) [inline customer-commons/src/main/java/no/unit/nva/customer/model/CustomerDao.java:91]: Minor whitespace issue after comma in parameter lists ;)

File: customer-commons/src/main/java/no/unit/nva/customer/model/CustomerDto.java [hunk 2]
```diff
@@ -221,7 +241,7 @@ public int hashCode() {
         return Objects.hash(getContext(), getId(), getIdentifier(), getCreatedDate(), getModifiedDate(), getName(),
                             getDisplayName(), getShortName(), getArchiveName(), getCname(), getInstitutionDns(),
                             getFeideOrganizationDomain(), getCristinId(), getCustomerOf(), getVocabularies(),
-                            getRorId(), getPublicationWorkflow());
+                            getRorId(), getPublicationWorkflow(),getDoiPrefix(),getDoiAgencyName());
```
Comment (ketilaa) [inline customer-commons/src/main/java/no/unit/nva/customer/model/CustomerDto.java:244]: Same minor whitespace issue ;)

File: customer-commons/src/main/java/no/unit/nva/customer/model/CustomerDto.java [hunk 3]
```diff
@@ -363,8 +385,20 @@ public Builder withPublicationWorkflow(PublicationWorkflow publicationWorkflow)
             return this;
         }
 
+        public Builder withDoiPreFix(String doiPrefix) {
+            customerDto.setDoiPrefix(doiPrefix);
+            return this;
+        }
+
+        public Builder withDoiName(String doiName) {
```
Comment (ketilaa) [inline customer-commons/src/main/java/no/unit/nva/customer/model/CustomerDto.java:393]: Why the naming inconsistency between builder and dto? doiName or doiAgencyName?

File: customer-commons/src/test/resources/features/get_customer.feature [hunk 4]
```diff
@@ -19,6 +19,7 @@ Feature: Get Customer
     And they see that the Customer has a created date
     And they see that the Customer has a modified date
     And they see that the Customer has a contact
+    And they see that the Customer has a doiPrefix
```
Comment (ketilaa) [inline customer-commons/src/test/resources/features/get_customer.feature:22]: Should it also include doiAgencyName?

File: customer/src/test/java/no/unit/nva/customer/get/GetCustomerHandlerTest.java [hunk 5]
```diff
@@ -36,6 +18,25 @@
 import org.junit.jupiter.api.Test;
 import org.zalando.problem.Problem;
 
+import java.io.ByteArrayOutputStream;
```
Comment (ketilaa) [inline customer/src/test/java/no/unit/nva/customer/get/GetCustomerHandlerTest.java:21]: Are you using the NVA code style? A lot of stuff was shifted around here.

**Review by ketilaa — CHANGES_REQUESTED** [review]

File: customer/src/test/java/no/unit/nva/customer/get/GetCustomerHandlerTest.java (hunk 5 above)
Comment (StigNorland) [reply customer/src/test/java/no/unit/nva/customer/get/GetCustomerHandlerTest.java:21]: Rurik hjalp meg med å sette opp rett CheckStyle, skal være i orden nå.

Men var det noe spesielt med akkurat denne linjen?

File: customer-commons/src/main/java/no/unit/nva/customer/model/CustomerDto.java [hunk 6]
```diff
@@ -363,6 +385,16 @@ public Builder withPublicationWorkflow(PublicationWorkflow publicationWorkflow)
             return this;
         }
 
+        public Builder withDoiPreFix(String doiPrefix) {
```
Comment (ketilaa) [inline customer-commons/src/main/java/no/unit/nva/customer/model/CustomerDto.java:388]: Be consistent with naming. Method say preFix, parameter prefix

File: customer-commons/src/main/java/no/unit/nva/customer/model/CustomerDao.java [hunk 7]
```diff
@@ -257,133 +279,145 @@ public boolean equals(Object o) {
...
         }
-        
+
         public Builder withPublicationWorkflow(PublicationWorkflow publicationWorkflow) {
             customerDb.setPublicationWorkflow(publicationWorkflow);
             return this;
         }
-        
+
+        public Builder withDoiPreFix(String doiPrefix) {
```
Comment (ketilaa) [inline customer-commons/src/main/java/no/unit/nva/customer/model/CustomerDao.java:411]: PreFix vs prefix

Comment (StigNorland) [reply customer-commons/src/main/java/no/unit/nva/customer/model/CustomerDao.java:411]: also gone....

File: customer-commons/src/main/java/no/unit/nva/customer/model/CustomerDao.java [hunk 8]
```diff
@@ -45,20 +45,22 @@ public class CustomerDao implements Typed {
     private String cname;
     private String institutionDns;
     private String feideOrganizationDomain;
+    private String doiPrefix;
+    private String doiAgencyName;
```
Comment (brinxmat) [inline customer-commons/src/main/java/no/unit/nva/customer/model/CustomerDao.java:49]: Før vi ruller ut bør disse være én objekt, gjerne i ny PR

File: customer-commons/src/main/java/no/unit/nva/customer/model/CustomerDto.java [hunk 9]
```diff
@@ -37,6 +37,8 @@ public class CustomerDto implements Context {
     private String cname;
     private String institutionDns;
     private String feideOrganizationDomain;
+    private String doiPrefix;
+    private String doiAgencyName;
```
Comment (brinxmat) [inline customer-commons/src/main/java/no/unit/nva/customer/model/CustomerDto.java:41]: Samme her

**Review by brinxmat — APPROVED** [review]

**Pull request closed without merging**